Components/LearningCore/Source/gov/sandia/cognition/learning/algorithm/ensemble/BaggingCategorizerLearner.java

```diff
@@ -119,5 +126,134 @@ protected void addEnsembleMember(
         this.ensemble.add(member, 1.0);
     }
 
+    @Override
+    public int[] getDataInBag()
```

Comment:
The answer is mostly for performance. It does look like it is exposing internals. However, its just a status report for what is in the current bag after the iteration so that things like the OOB stopping can be plugged in. If someone modified it then it wouldn't have any real bearing on the algorithm itself. And since this detail is exposed for this case, it doesn't make sense to waste the cycles copying it.

Often in the Foundry we expose some data like this and give people the freedom to use it. However, they also should be responsible with what they do with it. In this case, they'd only break their own code.

Anyway, in a case like this it is good to make it clearer in the javadoc that this is exposed without copying for performance so callers should be careful, so I'll add that.